services/sources/publisher_backlist_service.py

```diff
@@ -74,18 +74,28 @@ def delete_work(self, record_metadata_dict):
                     self.db_manager.session.query(Work).filter(Work.id == edition.work_id).delete()
                     es_work_resp.delete()
                     self.db_manager.session.commit()
-
+                else:
+                    self.delete_pub_backlist_edition_only(record.uuid_str, work)
         except Exception:
             logger.exception('Work/Edition does not exist or failed to delete work: {work.id}')
         finally:
             self.db_manager.session.close()
 
     def checkAllEditionsRelatedToRecord(self, record_uuid_str, work):
-        for edition in self.db_manager.session.query(Edition).filter(Edition.work_id == work.id):
+        for edition in self.db_manager.session.query(Edition).filter(Edition.work_id == work.id).all():
             if record_uuid_str not in edition.dcdw_uuids:
                 return False
         return True
```

Comment:
Based on the above query you're getting the first edition associated with the record uuid then grabbing the work. I think we can delete the work if that's the only edition. 

Would it be easier to just check if the length of editions is just equal to 1?

Reply:
You're right that checking the length of the editions would be easier